src/ai/backend/manager/scheduler/dispatcher.py

```diff
@@ -583,6 +587,29 @@ async def _update_session_status_data() -> None:
         if num_scheduled > 0:
             await self.event_producer.produce_event(DoPrepareEvent())
 
+    async def _filter_agent_by_container_limit(
+        self, candidate_agents: list[AgentRow]
+    ) -> list[AgentRow]:
+        raw_value = await self.shared_config.etcd.get(
+            "config/agent/max-container-count", scope=ConfigScopes.SGROUP
+        )
```

Comment:
We don't have to explicit specify the scope when _querying_ the configuration if we query the etcd from the agents, because `common.etcd` will automatically compose the values from multiple scopes based on the scopes initialized upon the agent startup.

So, to keep this preliminary filtering logic in the manager side, the agents should also store the hard-limit values in the Redis so that this function could fetch them at once to filter out.

Reply:
agents store their current container count to Redis.
Should I also store hard-limit value to Redis for this implementation?